### PR TITLE
Offline messaging

### PR DESCRIPTION
**Pull request by TalkTakesTime**
This should, I think, have everything requested by you @Zarel, so it should be ready to go whenever you deem suitable

### CODE REVIEW

**Comment by Joimer** [comment]
@Zarel 
I've had several instances on the last week in which this would have been useful to have :(

**Comment by Zarel** [comment]
I'll move it up my to-do list.

**Comment by TalkTakesTime** [comment]
Is this actually waiting on anything? I don't know about anyone else, but I'd really like to have this actually live because I find myself wanting to use it pretty often when I'm on, and I've heard the same from quite a few others

**Comment by Zarel** [comment]
It's waiting for a deep code review from me. I'll get to it right after I fix all the important bugs, I hope sometime today.

**Comment by TheImmortal** [comment]
I don't think it should be automatically sent; it should be a separate command and when you /pm an offline user, it can inform you of the /tell command. The reason for this is that you are PMing someone and they leave but you sent the message without noticing they left, or if you saw a PM too late and tried to respond but they left. The message you would say to them when they are on and when they are not is going to be different. The obvious example in the first case is that someone is leaving and you say "bye" too late, that shouldn't be saved to send to them later. For the second case, you were busy and got a PM from someone. When you look back, you respond with what you would say to them if they were still there. If they were offline and you wanted to respond, your message would most likely be something different beginning with "Sorry, I missed you" for example.

**Comment by TalkTakesTime** [comment]
That's a valid point @TheImmortal, and I'm inclined to agree with you, especially with the limited number of messages that can be queued at once. I'll implement that as soon as I have the chance

**Comment by kotarou3** [comment]
Some things to note:
 - One might want to be able to delete/modify queued tells such as their contents or target user
 - Delivery failure should be reported to the user (maybe as another tell?)

**Comment by TalkTakesTime** [comment]
@kotarou3 unless either of those are *mandatory* components, I think it would be better to implement them at a later date, as this has been hanging around for long enough.
It currently has all the things that I see as required and I believe all the things @Zarel requested when we discussed it (ages ago), and I personally think that having this go live earlier and those features added at a later date will be more beneficial than waiting until those features are implemented and up to standard as well before anything at all is live on PS

**Comment by TalkTakesTime** [comment]
I'm no longer going to update this when it stops being able to be automatically merged. If it's ever going to be added let me know then and I'll update it so it's able to be merged.

File: tells.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+ * specified in config.js
+ *
+ * @license MIT license
+ */
+
+var fs = require('fs');
+
+var tells = {inbox: {}, outbox: {}};
+try {
+	tells = JSON.parse(fs.readFileSync('config/tells.json'));
```
Comment (panpawn) [inline tells.js]: Was there anything wrong with this being a ``JSON``, @Zarel?

Comment (Zarel) [inline tells.js]: Well, it's not how I'd currently design it, but at the time it was fine.

**Pull request closed without merging**